dev/AppLifecycle/AppLifecycle.cpp

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include <pch.h>
+#include <AppLifecycle.h>
+#include <AppLifecycle.g.cpp>
+
+#include "LaunchActivatedEventArgs.h"
+#include "ProtocolActivatedEventArgs.h"
+#include "Shared.h"
+
+namespace winrt::Microsoft::ProjectReunion::implementation
+{
+    std::tuple<std::wstring, std::wstring> AppLifecycle::ParseCommandLine(std::wstring commandLine)
+    {
+        auto argsStart = commandLine.rfind(L"----") + 4;
+
+        // We explicitly use find_first_of here, so that the resulting data may contain : as a valid character.
+        auto argsEnd = commandLine.find_first_of(L":", argsStart);
+        auto argsLength = argsEnd - argsStart;
```

Comment:
is there reasonable way to do what intsafe.h does here?

Reply:
Elaborate please.

Reply:
subtraction has the potential for overflow.  intsafe has helpers that return an error in this case,